### PR TITLE
feat: add gatlingEnterpriseDeploy target HYB-515

### PR DESCRIPTION
**Pull request by bastien-gatling**
Motivation:
Make accessible new package descriptor feature

Modifications:
Add GatlingEnterpriseDeployTask

### CODE REVIEW

File: src/main/groovy/io/gatling/gradle/GatlingPlugin.groovy [hunk 1]
```diff
@@ -61,6 +64,13 @@ final class GatlingPlugin implements Plugin<Project> {
         }
     }
 
+    private void registerEnterpriseDeployTask(Project project, TaskProvider<GatlingEnterprisePackageTask> gatlingEnterprisePackageTask) {
+        project.tasks.register(ENTERPRISE_DEPLOY_TASK_NAME, GatlingEnterpriseDeployTask.class) {
+            inputs.files gatlingEnterprisePackageTask
+            dependsOn(gatlingEnterprisePackageTask)
```
Comment (Isammoc) [inline src/main/groovy/io/gatling/gradle/GatlingPlugin.groovy:70]: 1. not the only dependency I guess
2. does that answer my previous question? (I'm astonished to see this 2 informations so far from one another)

File: build.gradle [hunk 2]
```diff
@@ -9,14 +9,18 @@ java {
...
+  }
+}
 repositories {
   mavenCentral()
   gradlePluginPortal()
 }
 
 dependencies {
-  implementation "io.gatling:gatling-enterprise-plugin-commons:1.9.0-M11"
+  implementation "io.gatling:gatling-enterprise-plugin-commons:1.9.0-M20240321105048"
```
[the remark this answers is not in the document]
Comment (bastien-gatling) [reply build.gradle:23]: I guess we can wait the merge of the common' project PR, and release a M12 ;)

[the remark this answers is not in the document]
Comment (Isammoc) [reply build.gradle:23]: I don't mind the format. I'm triggered by the milestone. But that's on me

File: src/main/groovy/io/gatling/gradle/GatlingEnterpriseDeployTask.groovy [hunk 3]
```diff
@@ -0,0 +1,27 @@
+package io.gatling.gradle
+
+import io.gatling.plugin.BatchEnterprisePlugin
+import io.gatling.plugin.deployment.DeploymentConfiguration
+import io.gatling.plugin.model.Packager
+import org.gradle.api.DefaultTask
+import org.gradle.api.tasks.CacheableTask
+import org.gradle.api.tasks.TaskAction
+
+@CacheableTask
```
[the remark this answers is not in the document]
Comment (bastien-gatling) [reply src/main/groovy/io/gatling/gradle/GatlingEnterpriseDeployTask.groovy:10]: Yep, dependencies are described when you register task, as you seen in your following comment :)

File: src/main/groovy/io/gatling/gradle/GatlingPlugin.groovy (hunk 1 above)
Comment (bastien-gatling) [reply src/main/groovy/io/gatling/gradle/GatlingPlugin.groovy:70]: That's my only dependency for this task : I need gatlingEnterprisePackage executed, and the file produced (the simulation jar) is injected as input of my task.

Actually it's my understanding (and other task, like upload or enterpriseStart are implemented that way)

File: src/main/groovy/io/gatling/gradle/GatlingEnterpriseDeployTask.groovy [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    @TaskAction
+    void deploy() {
+        final GatlingPluginExtension gatlingPlugin = project.extensions.getByType(GatlingPluginExtension)
+        final String version = project.version.toString()
+        final BatchEnterprisePlugin enterprisePlugin = gatlingPlugin.enterprise.initBatchEnterprisePlugin(version, logger)
+
+        final File descriptorFile = DeploymentConfiguration.fromBaseDirectory(project.rootDir)
+        final File packageFile = inputs.files.singleFile
+        final Boolean isPrivateRepositoryEnabled = gatlingPlugin.enterprise.controlPlaneUrl != null
+        final String artefactId = project.name
```
Comment (Isammoc) [inline src/main/groovy/io/gatling/gradle/GatlingEnterpriseDeployTask.groovy:22]:
```suggestion
        final String artifactId = project.name
```

File: src/main/groovy/io/gatling/gradle/GatlingPlugin.groovy (hunk 1 above)
Comment (bastien-gatling) [reply src/main/groovy/io/gatling/gradle/GatlingPlugin.groovy:70]: After discussion : Cache disabled for this task

**Review by Isammoc — APPROVED** [review]

**Review by tpetillot — APPROVED** [review]

**Review by Obero — APPROVED** [review]

**Pull request merged**